homeassistant/components/cover/somfy_mylink.py

```diff
@@ -0,0 +1,234 @@
+        return self._is_opening
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return self._device_class
+
+    @property
+    def current_cover_position(self):
+        """Return the current position of the cover."""
+        if self._move_time:
+            return self._position
+
+    async def mylink_open_cover(self):
+        """Open the cover, respecting reversal status."""
+        if not self._reverse:
+            await self.somfy_mylink.move_up(self._target_id)
+        else:
+            await self.somfy_mylink.move_down(self._target_id)
+
+    async def mylink_close_cover(self):
+        """Close the cover, respecting reversal status."""
+        if not self._reverse:
+            await self.somfy_mylink.move_down(self._target_id)
+        else:
+            await self.somfy_mylink.move_up(self._target_id)
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        if self._state is not None:
+            return
+        state = await super().async_get_last_state()
+        if state:
+            self._state = state.state
+            self._state_ts = state.last_updated
+            if self._move_time:
+                self._position = state.attributes.get('current_position', 0)
+            else:
+                self._position = None
+            if state.state == 'closed':
+                self._closed = True
+            else:
+                self._closed = False
+
+    async def async_open_cover(self, **kwargs):
+        """Wrap Homeassistant calls to open the cover."""
+        if self._move_time:
+            await self.async_set_cover_position(position=100)
+            self._closed = False
+            return
+        await self.mylink_open_cover()
+        self._closed = None
+
+    async def async_close_cover(self, **kwargs):
+        """Wrap Homeassistant calls to close the cover."""
+        if self._move_time:
+            await self.async_set_cover_position(position=0)
+            self._closed = True
+            return
+        await self.mylink_close_cover()
+        self._closed = None
+
+    async def async_stop_cover(self, **kwargs):
+        """Stop the cover."""
+        await self.somfy_mylink.move_stop(self._target_id)
+        self._is_closing = False
+        self._is_opening = False
+        self._closed = None
+        if self._position is None:
+            return
+        if self._unsub_listener_cover is not None:
+            self._unsub_listener_cover()
+            self._unsub_listener_cover = None
+            self._set_position = None
+
+    async def async_set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        position = kwargs.get(ATTR_POSITION)
+        self._set_position = round(position, -1)
+        if self._position == position:
```

Comment:
We can never adjust the position if we have the wrong position information if we include this check. We shouldn't do that.